recipes/imgui/all/conanfile.py

```diff
@@ -21,11 +21,20 @@ class IMGUIConan(ConanFile):
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+        "enable_test_engine": [True, False]
     }
     default_options = {
         "shared": False,
         "fPIC": True,
+        "enable_test_engine": False
     }
+    
+    def _base_version(self):
+        match = re.match(r'cci\.\d{8}\+(?P<version>\d+\.\d+(?:\.\d+))\.docking', str(self.version)) or re.match(r'(?P<version>\d+\.\d+(.\d+)?)-docking', str(self.version))
+        return Version(match.group('version')) if match else self.version
+    
+    def _is_docking_branch(self):
+        return re.match(r'.+(\.|-)docking', str(self.version))
```

Comment:
```suggestion
    def _base_version(self):
        return self.version.split('-')[0]
    
    def _is_docking_branch(self):
        return "docking" in self.version
```

If this is the underlying goal, I would go for simplicity

Reply:
You're right, packages that used the cci versioning style aren't even in conandata.yml anymore. So no point in checking it with a complex regex. Similar for _is_docking_branch, although I would opt to go for .endswith("-docking"), but proposed solution also is fine for me. I'll push the needed changes asap.